In [2]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

#### Selecting out some common stars between MILES and CFLIB for comparison :



In [3]:
cflib_path = "../preprocessing/preprocess_cflib_vyom.npy"

cflib_flux = np.load(cflib_path)


Getting the names of the selected 850 stars in the CFLIB library

In [ ]:
cflib_preprocessed_stars = np.load("../preprocessing/cflib_data.npz")["fname"]

Making a dataframe of the miles data

In [8]:
df = pd.read_csv("miles_selected.dat")

In [9]:
df.head()

,Miles,Name,Teff,[Fe/H],logg
0,1,HD 224930,5411,-0.78,4.19
1,2,HD 225212,4179,0.18,0.85
2,3,HD 225239,5559,-0.51,3.72
3,4,HD 00004,6779,0.21,3.87
4,5,HD 00249,4766,-0.27,2.91


Stripping the star names off HD in the MILES dataframe for comparison

In [11]:
def stripping(value :str):
    if "HD " in value:
        return value.removeprefix("HD ")
    
df["Name"] = df["Name"].apply(stripping)

In [13]:
df.head()

,Miles,Name,Teff,[Fe/H],logg
0,1,224930,5411,-0.78,4.19
1,2,225212,4179,0.18,0.85
2,3,225239,5559,-0.51,3.72
3,4,00004,6779,0.21,3.87
4,5,00249,4766,-0.27,2.91


Finding the common stars between the selected CFLIB and MILES for comparison

In [16]:
count = 0
for i in df["Name"]:
    for j in cflib_preprocessed_stars:
        if i == j.removesuffix(".fits"):
            print(i)
            count +=1
    
print(count)    # 213



224930
225212
225239
10307
10700
10780
10975
13555
15596
15798
16673
17081
17361
19994
20893
22049
22484
23249
23841
24421
25329
26965
27295
27371
28305
28978
29139
30743
30834
31295
32147
33256
33276
35620
37160
37828
37984
38656
39853
39833
39801
41117
41692
41636
41597
42543
43042
43318
43380
44007
43947
45282
47205
48329
48433
48682
50420
51440
54810
54719
55575
57264
58207
58551
59984
59881
60179
61064
61935
62301
63302
63352
63791
65714
65900
67228
69897
72660
74395
75732
76151
76932
76813
78558
78209
79452
81192
81797
82210
83618
83506
84441
84737
85235
85503
86986
87141
87737
87822
88609
88737
88986
89010
89254
89449
89484
89707
89744
89822
90508
237903
91889
93487
95128
97907
99747
102224
102328
102870
105262
105546
106516
107113
107213
110897
112127
114038
114330
114710
114642
115383
115617
117176
117876
118055
118100
120136
121370
122563
123299
123657
124897
124850
125451
126141
127334
132345
134083
136064
136726
136202
139195
139641
139446
141004
141714
142373
143761
148786

Checking the following MILES website for cross reference between MILES ID and object name : 
[MILES data](http://svocats.cab.inta-csic.es/miles/index.php)

* HD 10307 -- 0066
* HD 10700 -- 0067
* HD 10975 -- 0070
* HD 13555 -- 0079
